### PR TITLE
Hq basic

### PR DESCRIPTION
**Pull request by tollesonpdx**
Added basic HQ functionality to generate only 5 miners, attempt to refine soup every term, and seek and destroy enemy bots. The net gun function will need some code.

### CODE REVIEW

File: src/team3player/RobotPlayer.java [hunk 1]
```diff
@@ -84,10 +84,24 @@ static void MinerStartup() {
     static void runHQ() throws GameActionException {
         if (firstTurn) {
             firstTurn = false;
-            HQMemory = new MemoryForHQ();
+            HQMemory = new team3player.MemoryForHQ();
+        }
+        for (RobotInfo bot : rc.senseNearbyRobots(-1, rc.getTeam().opponent())) {
+            if (rc.canShootUnit(bot.ID)) {
+                runNetGun();
```
Comment (KyleMcB) [inline src/team3player/RobotPlayer.java:91]: I would suggest rc.shootUnit(bot.ID) instead of runNetGun which would point to the logic of the net gun robot

**Review by KyleMcB — CHANGES_REQUESTED** [review]
This doesn't actually shoot down enemy drones. I left a comment on the line on one way you could do that.

File: src/team3player/RobotPlayer.java (hunk 1 above)
Comment (tollesonpdx) [reply src/team3player/RobotPlayer.java:91]: Yeah, the runNetGun function was already stubbed out in the template. I figured we'd put the shootUnit functionality within the runNetGun function at a later date. -- To that end I created a new card in Trello for the net gun code. The code at the HQ level just checks for enemy bots within range, and then checks to see if each of those bots are "shootable" using the given function canShootUnit. If the bot is shootable, it calls the runNetGun function that will contain the actual shooting code. I didn't dig too deep, but I'm guessing that we'll have to grab the bots coordinates and somehow feed it into the shoot function.

**Review by KyleMcB — APPROVED** [review]
We'll get the net gun working in a bit

**Pull request merged**